src/Line3d.ts

```diff
@@ -0,0 +1,185 @@
+import { Color, Euler, Group, Vector3 } from "three";
+import { IDrawableObject } from "./types";
+import { LineMaterial } from "three/addons/lines/LineMaterial";
+import { MESH_LAYER, OVERLAY_LAYER } from "./ThreeJsPanel";
+import { LineSegments2 } from "three/addons/lines/LineSegments2";
+import { LineSegmentsGeometry } from "three/addons/lines/LineSegmentsGeometry";
+
+const DEFAULT_VERTEX_BUFFER_SIZE = 1020;
+
+/**
+ * Simple wrapper for a 3D line segments object, with controls for vertex data,
+ * color, width, and segments visible.
+ */
```

Comment:
@frasercl I'm debating a little bit whether I should have TFE deal with all of this, and literally be passing in a LineSegments2 object? But I imagine the extra layer of abstraction is good?

Reply:
Perfect, thanks for the input!